[IV-24-25] Objetivo 6

MarioRgzLpz opened the pull request:
* [x] ¿Se han establecido criterios *a priori* para elegir el sistema de
integración continua más conveniente y se han calificado los sistemas que
cumplan esos requisitos según esos criterios?
* [x] ¿Se han examinado varios sistemas de integración continua?
* [x] ¿Se han configurado varios sistemas de integración continua?
* [x] ¿Se ha configurado correctamente el *Checks API* en los sistemas en los que
   sea necesario para que aparezca correctamente en GitHub (y se pueda comprobar
   desde los tests)?
* [x] ¿Uno de los sistemas configurados permite comprobar cuales son las versiones
   dellenguaje con las que funciona correctamente nuestra aplicación?
* [x] ¿Se escogen de forma adecuada las versiones del lenguaje que se testean,
   tanto en el sistema de CI como en el contenedor Docker?
* [x] ¿Se han justificado correctamente las versiones del lenguaje que se están
testeando y se ha comprobado que no se comprueban varias veces lo mismo (la
misma versión en CI y en el contenedor Docker en otro sistema CI?

MarioRgzLpz committed:
docs: closes #36

Añado el fichero con la elección de herramientas de CI

MarioRgzLpz committed:
feat: closes #38

Modifico el fichero iv.yaml añadiendo los ficheros de CI

MarioRgzLpz committed:
feat: completa #37

Añado el fichero de CI de github

MarioRgzLpz committed:
feat: completa #37

Añado el fichero de CI de gitlab

MarioRgzLpz committed:
feat: closes #37

Añado el fichero de CI de appveyor

MarioRgzLpz committed:
fix: completo #37

Arreglo github ci

MarioRgzLpz committed:
fix: completo #38

Arreglo iv.yaml

MarioRgzLpz committed:
fix: completo #38

Arreglo iv.yaml

MarioRgzLpz committed:
fix: completo #37

Arreglo fichero appveyor

MarioRgzLpz committed:
feat: closes #40

Añado el fichero Cargo.lock para arreglar el error

MarioRgzLpz committed:
feat: completo #37

Arrglo la CI con GitLab

MarioRgzLpz committed:
feat: arreglo #37

Modifico el CI de github para instalar cargo make y nextest como se indica en la documentación

MarioRgzLpz committed:
feat: completo #37

Añado distintas versiones de Rust para probar que todo funciona igual

MarioRgzLpz committed:
feat: completo #37

Añado Gitlab correctamente

MarioRgzLpz committed:
feat: completo #37

Añado Gitlab otra vez

MarioRgzLpz committed:
feat: completo #37

Añado Gitlab otra vez

MarioRgzLpz committed:
fix: arregla #40

Modifico el Dockerfile para comprobar que funciona

MarioRgzLpz committed:
fix: arregla #40

Elimino cambios en el dockerfile

MarioRgzLpz committed:
feat: completo #37

Modifico appveyor para que el comando sea el mismo que en los tests del repo de IV

MarioRgzLpz committed:
docs: completa #36

Modifico Gitlab como ci y añado circleci

MarioRgzLpz committed:
feat: completa #37

Añado distintas versiones de rust para los tests con la ci de github

MarioRgzLpz committed:
feat: completo #37

Añado la configuracion de circleci

MarioRgzLpz committed:
fix: completo #37

Elimino la configuración de gitlab porque han quitado la integración continua gratuita descrita en el tutorial

MarioRgzLpz committed:
fix: closes #40

Hago propietario al usuario test para que funcionen correctamente los tests

MarioRgzLpz committed:
fix: completo #37

Arreglo la integracion con circle

MarioRgzLpz committed:
fix: completo #37

Arreglo la integracion con github para usar las actions recomendadas

MarioRgzLpz committed:
fix: completo #40

Modifico Dockerfile para establecer el grupo test

JJ commented on Dockerfile:
```diff
@@ -9,7 +9,7 @@ LABEL maintainer="mariorgzlpz@correo.ugr.es" version="0.0.5"
 
 RUN adduser test \
-    && chown -R test /app/test
+    && chown -R test:test /app/test 
```
Esto no te va a comprar nada, y ya te lo dije desde el principio. Los permisos que vas a tener son los del directorio *montado* no los que tú le pongas.

MarioRgzLpz replied:
Resuelto aqui -> [c1249f4](https://github.com/MarioRgzLpz/ArbitrageBets/pull/39/commits/c1249f4caf991684287ba590d1e8c664782b87f6)

MarioRgzLpz committed:
feat: completo #37

Modifico la configuración de appveyor para evitar hacer siempre el pull de la imagen

MarioRgzLpz committed:
feat: completo #37

Modifico el usuario con el que ejecuta el contenedor appveyor

MarioRgzLpz committed:
fix: closes #40

Añado la configuración personalizada de cargo nextest para tener permisos de escritura

MarioRgzLpz committed:
feat: completo #40

Modifico el dockerfile para quitar algo innecesario

hossam1522 reviewed (commented):
Realmente no sé si he revisado demasiado rápido, porque además parece que hay fallo da fallo CircleCI, pero te dejo algunos comentarios ya que estoy.

MarioRgzLpz committed:
docs: completo #36

Añado el enlace que faltaba y añado el criterio por el cual he descartado Jenkins

JJ commented on docs/integracion-continua.md:
```diff
@@ -0,0 +1,32 @@
+## Selección de herramientas
+
+### [Jenkins](https://www.jenkins.io): 
+Es de codigo abierto, tiene gran cantidad de plugins que permite la integración con Github, además de que es compatible con Docker. Existen soluciones modernas para realizar lo mismo sin tener que instalarlo de manera local, por lo que queda descartado por esta razón.
```
Lo mejor es simplemente no incluirlo. Como tienes que instalártelo localmente, simplemente no cumple el criterio básico aquí que es que se pueda ejecutar online.

MarioRgzLpz replied:
Resuelto aqui -> [7fd9892](https://github.com/MarioRgzLpz/ArbitrageBets/pull/39/commits/7fd989228b3d76ca53faf98e5af8257ee4ad4d34)

MarioRgzLpz commented on .appveyor.yml:
```diff
@@ -0,0 +1,4 @@
+image: Ubuntu
+
+build_script:
+  - docker run -u 1001 -t -v `pwd`:/app/test mariorgzlpz/arbitragebets
```
Esto era ya que necesitaba correr desde una maquina sin permisos para probar el comando docker run porque me estaban dando problemas los tests. De todas formas he estado mirando y no he encontrado como usar directamente mi imagen de docker como image en appveyor. ¿Alguna guía de como realizar esto?

hossam1522 replied:
Entonces lo que se me ocurre es usar la imagen en CircleCI y en Appveyor hacerlo "manual".

MarioRgzLpz replied:
Pero manual ya lo hago en Github, por lo que no se cuanto sentido tenga el hacerlo así o si directamente podría eliminar la integracion con appveyor porque sería redundante.

hossam1522 replied:
Realmente podrías considerarlo así y dejar solo CircleCI y GitHub, no es mala alternativa. Aunque si tienes tiempo (cosa que igual no mucho xd) te recomendaría que lo intentases igualmente (para aprender más que nada, porque al final del día en GitHub lo haces instalando las cosas usando recursos de otros lados)

MarioRgzLpz replied:
Pues en estas fechas a los estudiantes no nos sobra el tiempo como tu bien sabrás. Voy a intentar algo y si veo que me lleva demasiado tiempo lo elimino.

MarioRgzLpz replied:
Despues de varias (muchas) horas entre research y integración he conseguido que funcione appveyor y para no ser redundante con lo que hace github he decidido ejecutar los tests en windows. Queda resuelto aquí -> [fb896b1](https://github.com/MarioRgzLpz/ArbitrageBets/pull/39/commits/fb896b1df5a4b07b53a49a98e0066afea624844f)

MarioRgzLpz committed:
docs: completo #36

Elimino criterio y jenkins

juanbarearojo reviewed (commented):
Hola he sido asignado para revisar tu pr. Además de lo que ha dicho hossam una cosa para que la tengas en cuenta en base al tema de la versiones. También digo que el último test de circle ci ha dado fallo.

MarioRgzLpz committed:
docs: #36

Añado justificacion de versiones

MarioRgzLpz commented on .github/workflows/github_rust_ci.yml:
```diff
@@ -0,0 +1,31 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: ["1.81.0", "1.82.0", "1.83.0"]
```
Resuelto aqui->[89947cc](https://github.com/MarioRgzLpz/ArbitrageBets/pull/39/commits/89947cc971d17dc18cc43fc38b725baed7fb478d)

MarioRgzLpz commented on docs/integracion-continua.md:
```diff
@@ -0,0 +1,32 @@
+Como bien hemos hablado [aquí]() para el correcto desarrollo de software es necesario emplear integración continua, por lo que debemos elegir alguna herramienta que nos ayude con esta tarea.
```
Resuelto aqui -> [9b94808](https://github.com/MarioRgzLpz/ArbitrageBets/pull/39/commits/9b94808fd03869a7ca5cb2c29e671b3c19434c0e)

MarioRgzLpz committed:
fix: completo #37

Corrijo versiones para evitar que se pisen

MarioRgzLpz commented on .github/workflows/github_rust_ci.yml:
```diff
@@ -0,0 +1,31 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: ["1.81.0", "1.82.0", "1.83.0"]
```
Cierto, acabo de comprobarlo. En el dockerfile siempre esta la ultima versión, aunque ahora mismo es la 1.83 porque tienen que actualizarlo pero para evitar que se pisen en github actions no pondre la última estable pero si las dos anteriores.
Corrijo esto aquí -> [5d86af5](https://github.com/MarioRgzLpz/ArbitrageBets/pull/39/commits/5d86af57eb166c89c9ec33b07d0c75279ac829db)

MarioRgzLpz committed:
fix: completo #37

Modifico la ruta del directorio a almacenar por nextest en circle ci

MarioRgzLpz committed:
fix: completo #37

Modifico la ruta del directorio siempre sin buscar ya que nextest.toml siempre esta

MarioRgzLpz committed:
fix: completo #37

Creo el directorio previamente para evitar el problema de permisos

MarioRgzLpz committed:
fix: completo #37

Modifico el directorio por tmp

MarioRgzLpz committed:
feat: completo #37

Modifico el fichero de appveyor para hacer una integracion manual y usando windows

MarioRgzLpz committed:
feat: completo #37

Elimino la comprobacion de versión de rust y cargo

MarioRgzLpz committed:
feat: completo #37

Elimino la instalacion de cargo make porque parece que esta instalado ya

MarioRgzLpz committed:
feat: completo #37

Elimino la instalacion de cargo nextest porque parece que esta instalado ya

MarioRgzLpz committed:
feat: completo #37

Añado la instalación de cargo make y nextest pero solo en caso de no estar presentes

MarioRgzLpz commented:
@hossam1522 @juanbarearojo Muchas gracias por los cambios sugeridos. Queda disponible de nuevo para revisión.

hossam1522 reviewed (approved):
Mis dieses.

MarioRgzLpz commented:
@JJ Como el tiempo brilla por su ausencia, pido revisión a pesar de que Juan todavia no ha aprobado lo que me ha sugerido, pero yo creo que esta bien :=).

JJ commented on .circleci/config.yml:
```diff
@@ -0,0 +1,21 @@
+      - run:
+          name: Establecer ruta para nextest
+          command: |
+            sed -i 's|dir =.*|dir = "/tmp/nextest"|' .config/nextest.toml
```
Esta es buena. Espero que entiendas lo que hace :-)

MarioRgzLpz replied:
La usamos en SWAP con Docker el cuatrimestre pasado y todavía lo tenia por ahí guardado en la cabeza.

JJ reviewed (approved):
:+1:
